src/screens/Exchange/Exchange.js

```diff
@@ -744,7 +777,10 @@ class ExchangeScreen extends React.Component<Props, State> {
     const isTakeOfferPressed = pressedOfferId === offerId;
     const isSetAllowancePressed = pressedTokenAllowanceId === offerId;
     const isShapeShift = offerProvider === PROVIDER_SHAPESHIFT;
-    const providerLogo = getProviderLogo(offerProvider);
+    const providerInfo = providersMeta.find(({ shim }) => shim === offerProvider) || {};
+    const { icon_large: providerIconPath } = providerInfo;
+    const providerLogoUri = providerIconPath && `${EXCHANGE_URL}/v2.0${providerIconPath}`;
+    const providerLogo = providerLogoUri ? { uri: providerLogoUri } : getProviderLogo(offerProvider);
```

Comment:
I'd extract this into a function, probably something like:

```js
const offerProviderLogo = (providerMeta: ProvidersMeta, offerProvider: string) => {
  const providerInfo = providersMeta.find(({ shim }) => shim === offerProvider);
  if (!providerInfo) {
    return getProviderLogo(offerProvider);
  }

  const { icon_large: providerIconPath } = providerInfo;
  return { uri: `${EXCHANGE_URL}/v2.0${providerIconPath}` };
};
```

Then

```js
const providerLogo = offerProviderLogo(providersMeta, offerProvider);
```

Reply:
Updated it here and on ExchangeConfirm